jina/serve/gateway.py

```diff
@@ -73,53 +74,13 @@ def __init__(
         self.grpc_tracing_server_interceptors = (
             self.runtime_args.grpc_tracing_server_interceptors
         )
-
-        import json
-
-        from jina.serve.streamer import GatewayStreamer, _ExecutorStreamer
-
-        graph_description = json.loads(self.runtime_args.graph_description)
-        graph_conditions = json.loads(self.runtime_args.graph_conditions)
-        deployments_addresses = json.loads(self.runtime_args.deployments_addresses)
-        deployments_metadata = json.loads(self.runtime_args.deployments_metadata)
-        deployments_no_reduce = json.loads(self.runtime_args.deployments_no_reduce)
-
-        self.streamer = GatewayStreamer(
-            graph_representation=graph_description,
-            executor_addresses=deployments_addresses,
-            graph_conditions=graph_conditions,
-            deployments_metadata=deployments_metadata,
-            deployments_no_reduce=deployments_no_reduce,
-            timeout_send=self.runtime_args.timeout_send,
-            retries=self.runtime_args.retries,
-            compression=self.runtime_args.compression,
-            runtime_name=self.runtime_args.runtime_name,
-            prefetch=self.runtime_args.prefetch,
+        self.logger = JinaLogger(self.name, **vars(self.runtime_args))
+        self._request_handler = GatewayRequestHandler(
```

Comment:
not sure I understand the motivation behind blending together streamer and executor into `request_handler` object.
I mostly see the Gateway unaffected by the upcoming refactoring in the sense that users who implement a custom gateway will own the serve and application/request handling layer.

I think since this class is exposed to the user, we should keep it intact and in case we need to adapt it to our case, better to have a jina specific `BaseGateway` to be used as the Base server